automate_latex_render/bb_interact_random_block.py

```diff
@@ -0,0 +1,162 @@
+    driver.get(url)
+    try:
+        login_button = driver.find_element(By.ID, 'topframe.login.label').click()
+        print('login found')
+        if hide:
+            login(driver)
+            assert 'Welcome' in driver.title, '\n\n     Login failed - check login details'
+        else:
+            print('Please enter your login details on the page')
+            print(driver.title)
+            WebDriverWait(driver, 300).until(EC.title_contains('Welcome'))
+        print(driver.title)
+        driver.get(url)
+    except NoSuchElementException as elem_error:
+        print('no login')
+        print('-----------------')
+        raise(elem_error)
+
+    try:
+        arrow_buttons = driver.find_elements(By.XPATH, "//*[@title='Options Menu: Question Text']")
+        q_nums = driver.find_elements(By.XPATH, "//*[starts-with(@id,'totalNoQuestions')]")
+        qs = [int(q.text) for q in q_nums]
+        cum_qs = np.cumsum(qs)
+        for i in range(len(arrow_buttons)):
+            time.sleep(1)
+            previews = driver.find_elements(By.XPATH, "//*[@class='itemHead']")
+            p = np.where(cum_qs>i)[0][0]
+            print('clicking: ',p, ' preview link')
+            driver.execute_script("arguments[0].scrollIntoView();", previews[p])
+            previews[p].click()
+
+            time.sleep(0.5)
+            arrow_buttons = driver.find_elements(By.XPATH, "//*[@title='Options Menu: Question Text']")
+            print('clicking on arrow')
+            arrow_buttons[i].click()
+            print('clicked on arrow')
+            edit_buttons = driver.find_elements(By.XPATH, "//*[starts-with(@id,'context_menu_tag')]")
+            #temp_adds = driver.find_elements(By.XPATH, "//*[@class='addBeforeLink']")
+
+            print('clicking on edit')
+            edit_buttons[0].click()
+            print('----PAUSE----')
+            time.sleep(1)
+            if randomise:
+                randomise_answers(driver, 'fRandomOrder')
+                randomise_answers(driver, 'randomOrderId')
+            try:
+                submit_button = driver.find_element(By.NAME, "bottom_Submit")
+            except:
+                print('failed to find submit, looking for next button')
+                next_button = driver.find_element(By.NAME, "bottom_Next")
+                print('found next button')
+                print('----PAUSE----')
+                time.sleep(1)
+                next_button.click()
+                print('clicked next button')
+                print('----PAUSE----')
+                time.sleep(1)
+                submit_button = driver.find_element(By.NAME, "bottom_Submit")
+               
+            submit_button.click()
+            print('Submitted question {}/{}'.format(i+1, len(arrow_buttons)))
+        arrow_buttons = driver.find_elements(By.XPATH, "//*[@title='Options Menu: Question Text']")
+
+    except Exception as e:
+        print("Unexpected error has occured")
+        print(e)
```

Comment:
The original traceback is probably more informative than the message that this gives. If you're catching this to make sure you close the driver then it's better to use `try/finally`.